apps/core/src/components/affine/setting-modal/index.tsx

```diff
@@ -65,6 +65,19 @@ export const SettingModal = ({
     onSettingClick({ activeTab: 'account', workspaceId: null });
   }, [onSettingClick]);
 
+  // Add ‘⌘+,’ shortcut keys as switches
+  useEffect(() => {
+    const keydown = (e: KeyboardEvent) => {
+      if ((e.key === ',' && e.metaKey) || (e.key === ',' && e.ctrlKey)) {
+        e.preventDefault();
+        setOpen(!open);
```

Comment:
setOpen(open => !open)

Reply:
There is an error in this way of writing

<img width="1180" alt="image" src="https://github.com/toeverything/AFFiNE/assets/33956589/359856f1-7570-4bbc-ae80-32950edc5252">

It may need to be modified synchronously here 👇

https://github.com/toeverything/AFFiNE/blob/a2e4ef904bf281d3c2a3969b1b7bf4b78a80a12b/apps/core/src/providers/modal-provider.tsx#L99-L104

Modify here too, if necessary

https://github.com/toeverything/AFFiNE/blob/3b6e145b234614d06c0b8dba3676944753d4ef1b/apps/core/src/components/pure/quick-search-modal/index.tsx#L57

Reply:
Yes, the upstream code should be updated same

Reply:
In fact, useEffect already depends on open. It will trigger an update after setOpen. So is it necessary to use the functional update approach to get the latest value?

Reply:
> In fact, useEffect already depends on open. It will trigger an update after setOpen. So is it necessary to use the functional update approach to get the latest value?

This is not supposed to be in react.js. `[open, setOpen]` will trigger this effect every time the value flips, this will cause extra unused re-hooks here